Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkComponent.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/RTTI/ReflectContext.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzToolsFramework/Viewport/LocalViewBookmarkComponent.h>
+
+namespace AzToolsFramework
+{
+    void LocalViewBookmarkComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<LocalViewBookmarkComponent>()->Field(
+                "LocalBookmarkFileName", &LocalViewBookmarkComponent::m_localBookmarksFileName);
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext
+                    ->Class<LocalViewBookmarkComponent>(
+                        "Local View Bookmark Component",
+                        "The Local View Bookmark Component allows the user to store bookmarks in a custom streg file.")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "Local View Bookmarks")
+                    ->Attribute(AZ::Edit::Attributes::AddableByUser, true)
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("Level", 0x9aeacc13))
+                    ->Attribute(AZ::Edit::Attributes::Category, "View Bookmarks")
+                    ->Attribute(AZ::Edit::Attributes::Icon, "Icons/Components/Comment.svg")
```

Comment:
todo - remember to switch to new icon. Have you asked UX?

Reply:
I don´t think an icon is necessary since the component cards won´t be exposed to the user. Right now all of this is there to make it easier to debug and develop for now but it will be removed in the final PR.